activesupport/lib/active_support/duration.rb

```diff
@@ -419,6 +426,29 @@ def sum(sign, time = ::Time.current)
         end
       end
 
+      def rounded_parts
+        parts = self.parts.dup
+        remainder = 0
+
+        PARTS.reverse.each do |part|
+          if remainder != 0
+            parts[part] ||= 0
+            parts[part] += remainder
+            remainder = 0
+          end
+
+          part_value = parts[part].to_i
+          part_modulo = TIME_MODULOS[part]
+
+          if part_value != 0 && part_modulo
+            remainder = part_value / part_modulo
+            parts[part] = part_value % part_modulo
+          end
+        end
+
+        parts
+      end
+
```

Comment:
The naming of the `remainder` var is confusing IMHO.
Given that it's storing the amount of a part that's moving to the bigger part, I think `overflow` would be a much clearer name, and the code would be easier to follow